src/radical/pilot/utils/component.py

```diff
@@ -361,6 +371,11 @@ def declare_publisher(self, topic, pubsub):
         self._log.debug('declared publisher : %s : %s : %s' \
                 % (topic, pubsub, q.name))
 
+        # FIXME: I am not exactly sure why this is 'needed', but declaring
+        #        a published as above and then immediately publishing on that
+        #        channel seems to sometimes lead to a loss of messages.
+        time.sleep(1)
+
```

Comment:
I get a bit nervous when we add another 500 lines to fix synchronisation and then ultimately it only works if we add a sleep ...